cmd/otel-allocator/watcher/promOperator.go

```diff
@@ -358,9 +392,96 @@ func (w *PrometheusCRWatcher) addStoreAssetsForPodMonitor(
 		if err = store.AddSafeAuthorizationCredentials(ctx, pmNamespace, endp.Authorization, smAuthKey); err != nil {
 			break
 		}
+
+		for _, rl := range endp.RelabelConfigs {
+			if rl.Action != "" {
+				if validateErr = validateRelabelConfig(*rl); validateErr != nil {
+					break
+				}
+			}
+		}
+
+		for _, rl := range endp.MetricRelabelConfigs {
+			if rl.Action != "" {
+				if validateErr = validateRelabelConfig(*rl); validateErr != nil {
+					break
+				}
+			}
+		}
 	}
 
 	if err != nil {
 		w.logger.Error(err, "Failed to obtain credentials for a PodMonitor", "podMonitor", pmName)
 	}
+
+	if validateErr != nil {
+		return validateErr
+	}
+
+	return nil
+}
+
+// validateRelabelConfig validates relabel config for service and pod monitor,
+// based on the service monitor and pod metrics endpoints specs.
+// This code borrows from
+// https://github.com/prometheus-operator/prometheus-operator/blob/ba536405154d18f3a6f312818283d671182af6f3/pkg/prometheus/resource_selector.go#L237
+func validateRelabelConfig(rc monitoringv1.RelabelConfig) error {
```

Comment:
Would it be useful for us to reach out to the prometheus operator folks to see if they would expose this method?

Reply:
Thanks @jaronoff97. I have reached out to them, waiting for a response,(https://github.com/prometheus-operator/prometheus-operator/pull/6057#issuecomment-1804829543)  based on that I can add the tests.

Reply:
Also created an issue here - https://github.com/prometheus-operator/prometheus-operator/issues/6076

Reply:
@jaronoff97 - There is a question about breaking changes. I have answered it from what I understand and what I think the way forward would be. Could you please respond to provide your perspective and correct me if i am wrong?
https://github.com/prometheus-operator/prometheus-operator/issues/6076#issuecomment-1808719873

Reply:
yep, i responded there :) thanks for making the issue. Let's wait to hear from them before merging this – i would love to just use their method rather than having to write it ourselves.